src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs

```diff
@@ -83,16 +83,16 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
             return state;
         }
 
-        private async Task<NuGetReference?> GetUpdatedPackageVersionAsync(NuGetReference packageReference, IEnumerable<TargetFrameworkMoniker> targetFramework, CancellationToken token)
+        private async Task<NuGetReference?> GetUpdatedPackageVersionAsync(NuGetReference packageReference, IEnumerable<TargetFrameworkMoniker> tfms, CancellationToken token)
         {
-            var latestMinorVersions = await _packageLoader.GetNewerVersionsAsync(packageReference, true, token).ConfigureAwait(false);
+            var latestMinorVersions = await _packageLoader.GetNewerVersionsAsync(packageReference, tfms, true, token).ConfigureAwait(false);
             NuGetReference? prereleaseCandidate = null;
 
             foreach (var newerPackage in latestMinorVersions)
             {
-                if (await _packageLoader.DoesPackageSupportTargetFrameworksAsync(newerPackage, targetFramework, token).ConfigureAwait(false))
+                if (await _packageLoader.DoesPackageSupportTargetFrameworksAsync(newerPackage, tfms, token).ConfigureAwait(false))
```

Comment:
Can this `if` go away now that `GetNewerVersionsAsync` filters for TFMs?

Reply:
Maybe, but I'll keep that for now and we can check on another PR